modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMStructureLocalServiceImpl.java

```diff
@@ -1562,6 +1613,11 @@ public DDMStructure updateStructure(
 		structure.setUserId(userId);
 		structure.setParentStructureId(parentStructureId);
 
+		_validateExternalReferenceCode(
+			externalReferenceCode, groupId, classNameId);
```

Comment:
Add FF and move this to before this line:

`User user = _userLocalService.getUser(userId);`